examples/text_summarization/bertsum_cnndm.ipynb

```diff
@@ -0,0 +1,3852 @@
+       " 3554,\n",
+       " 1996,\n",
+       " 6077,\n",
+       " 1010,\n",
+       " 2021,\n",
+       " 1996,\n",
+       " 4861,\n",
+       " 2056,\n",
+       " 2002,\n",
+       " 2106,\n",
+       " 2025,\n",
+       " 6655,\n",
+       " 2006,\n",
+       " 1996,\n",
+       " 102]"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bert_format_data[5]['src']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"new : nfl chief , atlanta falcons owner critical of michael vick 's conduct .<q>nfl suspends falcons quarterback indefinitely without pay .<q>vick admits funding dogfighting operation but says he did not gamble .<q>vick due in federal court monday ; future in nfl remains uncertain .<q>\""
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bert_format_data[5]['tgt_txt']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0]"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bert_format_data[5]['labels']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['( cnn ) -- the national football league has indefinitely suspended atlanta falcons quarterback michael vick without pay , officials with the league said friday .',\n",
+       " 'nfl star michael vick is set to appear in court monday .',\n",
+       " 'a judge will have the final say on a plea deal .',\n",
+       " 'earlier , vick admitted to participating in a dogfighting ring as part of a plea agreement with federal prosecutors in virginia . ``',\n",
+       " 'your admitted conduct was not only illegal , but also cruel and reprehensible .',\n",
+       " 'your team , the nfl , and nfl fans have all been hurt by your actions , `` nfl commissioner roger goodell said in a letter to vick .',\n",
+       " 'goodell said he would review the status of the suspension after the legal proceedings are over .',\n",
+       " 'in papers filed friday with a federal court in virginia , vick also admitted that he and two co-conspirators killed dogs that did not fight well .',\n",
+       " \"falcons owner arthur blank said vick 's admissions describe actions that are `` incomprehensible and unacceptable . ``\",\n",
+       " 'the suspension makes `` a strong statement that conduct which tarnishes the good reputation of the nfl will not be tolerated , `` he said in a statement .',\n",
+       " \"watch what led to vick 's suspension `` go
```

Comment:
why do we need to expose the device_id?

Reply:
for machine with multiple devices, this actually provides users the capability of running multiple notebooks at the same time.

Reply:
can they use multiple GPUs for one training process?